Dynamically extend GPU Process to Apple Camera Client when appropriate

brentfulgham opened the pull request:
#### a782040e52f01520625c8fc1c36565e3e777eeab
<pre>
Dynamically extend GPU Process to Apple Camera Client when appropriate
<a href="https://bugs.webkit.org/show_bug.cgi?id=256013">https://bugs.webkit.org/show_bug.cgi?id=256013</a>
&lt;rdar://108282689&gt;

Reviewed by Eric Carlson.

Extend a mach connection to an Apple Camera User Client service when appropriate.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::addCameraSandboxExtensions):
* Source/WebKit/UIProcess/UserMediaProcessManager.cpp:
(WebKit::UserMediaProcessManager::willCreateMediaStream):
(WebKit::UserMediaProcessManager::revokeSandboxExtensionsIfNeeded):

Canonical link: <a href="https://commits.webkit.org/263469@main">https://commits.webkit.org/263469@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a77daa810df3ae16e9b185ecfc43e791f0b8bea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4703 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4823 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4983 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6207 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4855 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4973 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4789 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5082 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4780 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4868 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4211 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6217 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/4783 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2357 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4202 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/9187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/3908 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4216 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/4276 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5847 "263 api tests failed or timed out") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4471 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4684 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3820 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4798 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4208 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1185 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1152 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8256 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4920 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4568 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1296 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
```diff
@@ -243,7 +244,15 @@ static inline bool addCameraSandboxExtensions(Vector<SandboxExtension::Handle>&
-            extensions.append(WTFMove(*appleCameraServicePathSandboxExtensionHandle));
+#if HAVE(APPLE_CAMERA_USER_CLIENT)
+            // Needed for rdar://108282689:
+            auto appleCameraUserClientExtensionHandle = SandboxExtension::createHandleForMachLookup("com.apple.aneuserd"_s, std::nullopt);
```
Nit: can you use `appleCameraUserClientPath` instead of the string constant?

eric-carlson reviewed (approved):
r=me

brentfulgham committed:
Dynamically extend GPU Process to Apple Camera Client when appropriate
https://bugs.webkit.org/show_bug.cgi?id=256013
<rdar://108282689>

Reviewed by Eric Carlson.

Extend a mach connection to an Apple Camera User Client service when appropriate.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::addCameraSandboxExtensions):
* Source/WebKit/UIProcess/UserMediaProcessManager.cpp:
(WebKit::UserMediaProcessManager::willCreateMediaStream):
(WebKit::UserMediaProcessManager::revokeSandboxExtensionsIfNeeded):

Canonical link: https://commits.webkit.org/263469@main

pvollan commented on Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
```diff
@@ -700,7 +700,9 @@
-    (iokit-user-client-class "AudioAUUC"))
+    (iokit-user-client-class "AudioAUUC")
+    ;; Following is needed due to <rdar://problem/108282689>
+    (iokit-user-client-class "H11ANEInDirectPathClient"))
```
Could this IOKit class also be tied to an extension?
